frame/contracts/src/lib.rs

```diff
@@ -549,10 +554,13 @@ decl_module! {
 			let origin = ensure_signed(origin)?;
```

Comment:
those early returns are still charged for the MaxCodeSize but I think it is fine.

Reply:
Yes. They are even charged for the whole gas meter in case of an early error. Those errors are all in the control of the caller to avoid. That said, I will try to eliminate overcharging in the future but for now I think this is fine.